components/datetime/src/format/datetime.rs

```diff
@@ -67,7 +67,7 @@ use writeable::{Part, Writeable};
 /// ```
 #[derive(Debug, Copy, Clone)]
 pub struct FormattedDateTime<'l> {
```

Comment:
Note: `FormattedDateTime` is being deleted in #5543. But, it looks like this PR is mostly independent of that.

Reply:
I had this on top of #5543 but I don't want to block on it because it doesn't seem ready.

Reply:
I wanted to finish it on Friday but ran out of time... now it will be bitrot. Oh well :)